src/GraphQL/GraphQLBuilderExtensions.cs

```diff
@@ -219,6 +220,100 @@ public static IGraphQLBuilder AddSchema<TSchema>(this IGraphQLBuilder builder, F
+            where TGraphTypeMappingResolver : class, IGraphTypeMappingResolver
+        {
+            builder.Services.Register<IGraphTypeMappingResolver, TGraphTypeMappingResolver>(ServiceLifetime.Singleton);
+            return builder;
+        }
+
+        /// <summary>
+        /// Registers an instance of <typeparamref name="TGraphTypeMappingResolver"/> with the dependency injection
+        /// framework as a singleton of type <see cref="IGraphTypeMappingResolver"/> using the specified factory delegate.
+        /// <br/><br/>
+        /// An <see cref="IGraphTypeMappingResolver"/> can be used to map one or more CLR types to graph types.
+        /// For instance, unmapped CLR output types can be mapped to <see cref="AutoRegisteringObjectGraphType{TSourceType}"/>
+        /// types.
+        /// </summary>
+        public static IGraphQLBuilder AddGraphTypeMappingResolver<TGraphTypeMappingResolver>(this IGraphQLBuilder builder, Func<IServiceProvider, TGraphTypeMappingResolver> factory)
+            where TGraphTypeMappingResolver : class, IGraphTypeMappingResolver
+        {
+            builder.Services.Register<IGraphTypeMappingResolver>(factory, ServiceLifetime.Singleton);
+            return builder;
+        }
+
+        /// <summary>
+        /// Registers an instance of <typeparamref name="TGraphTypeMappingResolver"/> with the dependency injection
+        /// framework as a singleton of type <see cref="IGraphTypeMappingResolver"/> using the specified instance.
+        /// <br/><br/>
+        /// An <see cref="IGraphTypeMappingResolver"/> can be used to map one or more CLR types to graph types.
+        /// For instance, unmapped CLR output types can be mapped to <see cref="AutoRegisteringObjectGraphType{TSourceType}"/>
+        /// types.
+        /// </summary>
+        public static IGraphQLBuilder AddGraphTypeMappingResolver<TGraphTypeMappingResolver>(this IGraphQLBuilder builder, TGraphTypeMappingResolver instance)
+            where TGraphTypeMappingResolver : class, IGraphTypeMappingResolver
+        {
+            builder.Services.Register<IGraphTypeMappingResolver>(instance);
+            return builder;
+        }
+        #endregion
+
+        #region - AddAutoSchema / WithMutation / WithSubscription -
+        /// <summary>
+        /// Registers an instance of the <see cref="Schema"/> class within the dependency injection framework as a singleton.
+        /// <see cref="ISchema"/> is also registered if it is not already registered within the dependency injection framework.
+        /// <see cref="Schema.Query"/> is set to an instance of <see cref="AutoRegisteringObjectGraphType{TSourceType}"/> with
+        /// <typeparamref name="TQueryClrType"/> as TSourceType.
+        /// <br/><br/>
+        /// Additionally, this method calls <see cref="AddAutoClrMappings(IGraphQLBuilder, bool, bool)">AddAutoClrMappings</see>
```

Comment:
Technically it does not call AddAutoClrMappings.

Reply:
True, but for the purposes of the documentation, it does.